core/tool_roundtrip.c

```diff
@@ -567,6 +567,65 @@ _roundtrip_put_from_file (const char * const * properties)
 	oio_error_pfree (&err);
 }
 
+/* Upload a content larger than the chunk size
+ * then check the chunks' size */
+static void
+check_chunksize (const char * const * properties)
+{
+	const gsize chunk_size = 5*1024*1024;
+	struct oio_error_s *err = NULL;
+
+	struct oio_url_s *url_random = oio_url_dup(url);
+	gchar path[32];
+	oio_str_randomize(path, oio_ext_rand_int_range(7,32), random_chars);
+	oio_url_set(url_random, OIOURL_PATH, path);
+
+	/* generate a buffer to work in. First we fill it of random bytes */
+	gsize size = 7*1024*1024;
+	guint8 *buffer = g_malloc(size);
+	g_assert_nonnull(buffer);
+	oio_buf_randomize(buffer, size);
+	gchar *pre_hash =
+		g_compute_checksum_for_data(G_CHECKSUM_MD5, buffer, size);
+	g_assert_nonnull(pre_hash);
+	oio_str_upper(pre_hash);
+	if (GRID_TRACE2_ENABLED()) {
+        g_file_set_contents("/tmp/pre", (gchar *) buffer, size, (GError **) &err);
+        NOERROR(err);
+	}
+
+	struct oio_sds_ul_dst_s ul_dst = OIO_SDS_UPLOAD_DST_INIT;
+	ul_dst.url = url;
+	ul_dst.autocreate = 1;
+	ul_dst.append = 0;
+	ul_dst.out_size = 0;
+	ul_dst.properties = properties;
+	ul_dst.chunk_size = chunk_size;
+	err = oio_sds_upload_from_buffer(client, &ul_dst, buffer, size);
+	NOERROR(err);
+
+	/* get details on the content */
+	gsize max_size[2] = {0};
+	void _on_metachunk(void *i UNUSED, guint seq UNUSED, gsize offt, gsize len) {
+		GRID_DEBUG("metachunk: %u, %"G_GSIZE_FORMAT" %"G_GSIZE_FORMAT,
+				seq, offt, len);
+		max_size[seq] = len;
```

Comment:
This is very dangerous. If for some reason the chunking is broken, and generates small chunks, this will cause a segmentation fault, instead of failing the assertion below.